__mocks__/FileCard.test.js

```diff
@@ -0,0 +1,135 @@
+}
+
+const mockSetPreview = jest.fn();
+
+jest.mock('react', () => ({
+    ...jest.requireActual('react'),
+    useState: (initialState) => [initialState, mockSetPreview]
+}))
+
+test('FileForm PropTypes', () => {
+    const conformingProps = {
+        repository: {
+            owner: {
+                username: 'username',
+                avatar_url: 'avatar_url',
+            },
+            name: 'name',
+            avatar_url: 'avatar_url',
+            permissions: {
+                push: true,
+            },
+            full_name: 'full_name',
+            default_branch: 'default_branch'
+        },
+        file:{
+            name: 'file.name',
+            path: 'file.path',
+            sha: 'file.sha',
+            content: 'file.content',
+            branch: 'file.branch',
+            filepath: 'filepath',
+            save: () => {},
+            dangerouslyDelete: () => {},
+            close: () => {}
+        },
+        isAuthenticated: true,
+    }
+    checkProps(FileCard, conformingProps);
+});
+
+// render tests
+describe('render fileCard elements',() => {
+    test('render fileCard', () => {
+        const wrapper = setupWrapper();
+        const fileCard = findByAttribute(wrapper, 'component-fileCard');
+        expect(fileCard.length).toBe(1);
+    });
+    
+    test('render cardHeader', () => {
+        const wrapper = setupWrapper();
+        const cardHeader = findByAttribute(wrapper, 'cardHeader');
+        expect(cardHeader.length).toBe(1);
+    });
+    
+    test('render blockEditable', () => {
+        const wrapper = setupWrapper();
+        const blockEditable = findByAttribute(wrapper, 'blockEditable');
+        expect(blockEditable.length).toBe(1);
+    });
+    
+    test('render previewButton', () => {
+        const wrapper = setupWrapper();
+        const previewButton = findByAttribute(wrapper, 'previewButton');
+        expect(previewButton.length).toBe(1);
+    });
+    
+    test('render saveButton', () => {
+        const wrapper = setupWrapper();
+        const saveButton = findByAttribute(wrapper, 'saveButton');
+        expect(saveButton.length).toBe(1);
+    });
+
+    test('render deleteButton', () => {
+        const wrapper = setupWrapper();
+        const deleteButton = findByAttribute(wrapper, 'deleteButton');
+        expect(deleteButton.length).toBe(1);
+    });
+
+    test('render closeButton', () => {
+        const wrapper = setupWrapper();
+        const closeButton = findByAttribute(wrapper, 'closeButton');
+        expect(closeButton.length).toBe(1);
+    });
+});
+
+test('switch the preview state', () => {
+    const wrapper = setupWrapper();
+    const blockEditable = findByAttribute(wrapper, 'blockEditable');
+    expect(blockEditable.props().preview).toBeTruthy();
+    const previewButton = findByAttribute(wrapper, 'previewButton');
+    previewButton.simulate('click');
+    expect(mockSetPreview).toBeCalledWith(false);
+    // NEED HELP
```

Comment:
@ancientTexts-net 
in this function I am testing the initial value to be `true` and the and when the user click on the button I test that that the `setPreview` is call with the `false` as param. which describe the switch from true to false. 
but I don't know how to set realy the state on the component to  `false` so I will be able to test the switch from false to true.
I found that in a class component we use something like this 
  `wrapper.instance().setPreview(false)`
I am looking for an equivalent for the hooks